torch/csrc/jit/script/sugared_value.h

```diff
@@ -404,6 +416,64 @@ struct TORCH_API IsInstanceValue : SugaredValue {
   }
 };
 
+// matched against for special handling of range expressions
+struct TORCH_API RangeValue : SugaredValue {
+  RangeValue(const SourceRange& loc, Function&m, std::vector<Value*> inputs);
+  std::string kind() const override {
+    return "range";
+  }
+  Value* len(const SourceRange& loc, Function& m) override;
+  Value* get_elem(const SourceRange&loc, Function& m, Value* i) override;
+
+  private:
+    Value* start_;
+    Value* end_;
+    Value* step_;
+    bool is_simple_;
```

Comment:
What does this mean?

Reply:
`is_simple_` means we the input args of `range` only contains `end`, and this serves as a flag in `len` and `getelem` to not insert unnecessary builtin nodes.

Reply:
maybe use `has_start_and_step_` then?